src/github/webhook.py

```diff
@@ -15,10 +15,10 @@
 def _handle_pull_request_webhook(payload: dict) -> HttpResponse:
     pull_request_id = payload["pull_request"]["node_id"]
     pull_request = graphql_client.get_pull_request(pull_request_id)
+    # a label change will trigger this webhook, so it may trigger automerge
+    github_logic.maybe_automerge_pull_request_and_rerun_stale_checks(pull_request)
+    github_logic.maybe_add_automerge_warning_comment(pull_request)
```

Comment:
I actually think we want to keep these components inside the lock. These don't mutate the task but it could mutate the PR.

Same with the other moves